plugins/inputs/postgresql_extensible/postgresql_extensible_test.go

```diff
@@ -293,9 +293,9 @@ func (f fakeRow) Scan(dest ...interface{}) error {
 	}
 
 	for i, d := range dest {
-		switch d.(type) {
+		switch d := d.(type) {
 		case (*interface{}):
-			*d.(*interface{}) = f.fields[i]
+			*d = f.fields[i]
```

Comment:
Shouldn't we use a simple if/else statement here and avoid the whole switch clutter?

Reply:
Hi @srebhan 

We definitely can! The DeepSource checks have flagged all the issues related to this [here](https://deepsource.io/gh/HarshCasper/telegraf/issue/CRT-A0014/occurrences)

For example, a Switch statement like this: 

```
switch x := x.(type) {
case int:
    body()
}
```

can be written as: 

```
if x, ok := x.(int); ok {
    body()
}
``` 

The only issue here would be their occurrence a lot of times in a codebase. So shall I clean all of them up in this PR or shall I raise a separate PR for all of this cleanup?

Reply:
I think you should clean them in one go. I'm on some other linter errors and it's a mess as the code base changes often and you will touch a lot of places. So it might make sense to split the PR, but it's not a must. Be prepared to rebase often! ;-) You have been warned. :-)